cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java

```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
      * [NGSIMongoSink.configure] -------- Configured 'collection_prefix' is encoded when having forbiden characters.
      */
     @Test
-    public void testConfigureCollectionPrefixIsEncoded() {
+    public void testConfigureCollectionPrefixIsEncodedOldEncoding() {
         System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
                 + "-------- Configured 'collection_prefix' is encoded when having forbiden characters");
         String collectionPrefix = "this\\is/a$prefix.with-forbiden,chars:-.";
         String dbPrefix = "sth_";
-        String dataModel = null; // defaulting
+        String dataModel = null; // default
+        String enableEncoding = "false";
         NGSIMongoBaseSinkImpl sink = new NGSIMongoBaseSinkImpl();
-        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel));
-        String encodedCollectionPrefix = NGSIUtils.encodeSTHCollection(collectionPrefix);
+        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel,
+                enableEncoding));
+        String expectedCollectionPrefix = NGSIUtils.encodeSTHCollection(collectionPrefix);
         
         try {
-            assertTrue(sink.collectionPrefix.equals(encodedCollectionPrefix));
+            assertEquals(expectedCollectionPrefix, sink.collectionPrefix);
             System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
                     + "-  OK  - 'collection_prefix=" + collectionPrefix
-                    + "' correctly encoded as '" + encodedCollectionPrefix + "'");
+                    + "' correctly encoded as '" + expectedCollectionPrefix + "'");
         } catch (AssertionError e) {
             System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
                     + "- FAIL - 'collection_prefix=" + collectionPrefix
-                    + "' wrongly encoded as '" + encodedCollectionPrefix + "'");
+                    + "' wrongly encoded as '" + expectedCollectionPrefix + "'");
             throw e;
-        } // try catch
-    } // testConfigureCollectionPrefixIsEncoded
+        } // try catch // try catch
+    } // testConfigureCollectionPrefixIsEncodedOldEncoding
+    
+    /**
+     * [NGSIMongoSink.configure] -------- Configured 'collection_prefix' is encoded when having forbiden characters.
+     */
+    @Test
+    public void testConfigureCollectionPrefixIsEncodedNewEncoding() {
+        System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
+                + "-------- Configured 'collection_prefix' is encoded when having forbiden characters");
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4